### PR TITLE
Publish v2.2.0

### PR DESCRIPTION
**Pull request by JakeHartnell**
There are no functionality changes in this release, just renaming of packages (because of name collisions in crates.io) as well as moving stuff around to reduce circular dependencies so we can publish all the DAO DAO packages. Also fixes up CI as nightly changes have broken it.

Closes https://github.com/DA0-DA0/dao-contracts/issues/242. Finally. :partying_face:

NOTE: have tested the publish script, will tag `v2.2.0` last after getting this reviewed.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -64,46 +64,46 @@ wynd-utils = "0.4.1"
...
+dao-proposal-single = { path = "./contracts/proposal/dao-proposal-single", version = "2.1.3" }
+dao-proposal-sudo = { path = "./test-contracts/dao-proposal-sudo", version = "2.1.3" }
+dao-testing = { path = "./packages/dao-testing", version = "2.1.3" }
+dao-vote-hooks = { path = "./packages/dao-vote-hooks", version = "2.1.3" }
+dao-voting = { path = "./packages/dao-voting", version = "2.1.3" }
+dao-voting-cw20-balance = { path = "./test-contracts/dao-voting-cw20-balance", version = "2.1.3" }
+dao-voting-cw20-staked = { path = "./contracts/voting/dao-voting-cw20-staked", version = "2.1.3" }
+dao-voting-cw4 = { path = "./contracts/voting/dao-voting-cw4", version = "2.1.3" }
+dao-voting-cw721-staked = { path = "./contracts/voting/dao-voting-cw721-staked", version = "2.1.3" }
+dao-voting-native-staked = { path = "./contracts/voting/dao-voting-native-staked", version = "2.1.3" }
```
Comment (JakeHartnell) [inline Cargo.toml:97]: Wildcard `"*"` versions are not allowed for `cargo publish`.

File: Cargo.toml [hunk 2]
```diff
@@ -64,46 +64,46 @@ wynd-utils = "0.4.1"
...
+cw-core-v1 = { package = "cw-core", version = "0.1.0" }
+cw-proposal-single-v1 = { package = "cw-proposal-single", version = "0.1.0" }
 cw-utils-v1 = {package = "cw-utils", version = "0.13"}
-cw20-stake-external-rewards-v1 = { package = "stake-cw20-external-rewards", version = "0.2.6", git = "https://github.com/DA0-DA0/dao-contracts.git", tag = "v1.0.0" }
-cw20-stake-reward-distributor-v1 = { package = "stake-cw20-reward-distributor", version = "0.1.0", git = "https://github.com/DA0-DA0/dao-contracts.git", tag = "v1.0.0" }
-cw20-stake-v1 = { package = "cw20-stake", version = "0.2.6", git = "https://github.com/DA0-DA0/dao-contracts.git", tag = "v1.0.0" }
-cw20-staked-balance-voting-v1 = { package = "cw20-staked-balance-voting", version = "0.1.0", git = "https://github.com/DA0-DA0/dao-contracts.git", tag = "v1.0.0" }
-cw4-voting-v1 = { package = "cw4-voting", version = "0.1.0", git = "https://github.com/DA0-DA0/dao-contracts.git", tag = "v1.0.0" }
-voting-v1 = { package = "voting", version = "0.1.0", git = "https://github.com/DA0-DA0/dao-contracts.git", tag = "v1.0.0" }
```
Comment (JakeHartnell) [inline Cargo.toml:108]: Using git is not allowed for publishing packages to crates.io

Published these in the `v1.0.1` release: https://github.com/DA0-DA0/dao-contracts/releases/tag/v1.0.1

File: contracts/dao-dao/Cargo.toml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "dao-core"
+name = "dao-dao"
```
Comment (JakeHartnell) [inline contracts/dao-dao/Cargo.toml:2]: Unfortunately `dao-core` was taken on crates.io.

File: Cargo.toml [hunk 4]
```diff
@@ -64,46 +64,46 @@ wynd-utils = "0.4.1"
...
+cw20-stake-external-rewards-v1 = { package = "stake-cw20-external-rewards", version = "0.2.6" }
+cw20-stake-reward-distributor-v1 = { package = "stake-cw20-reward-distributor", version = "0.1.0" }
+cw20-stake-v1 = { package = "cw20-stake", version = "0.2.6" }
+cw20-staked-balance-voting-v1 = { package = "cw20-staked-balance-voting", version = "0.1.0" }
+cw4-voting-v1 = { package = "cw4-voting", version = "0.1.0" }
+voting-v1 = { package = "dao-voting", version = "0.1.0" }
 stake-cw20-v03 = { package = "stake-cw20", version = "0.2.6", git = "https://github.com/DA0-DA0/dao-contracts.git", tag = "v0.3.0" }
```
Comment (NoahSaso) [inline Cargo.toml:109]: Publish legacy contract and remove git reference?

File: contracts/README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # DAO Contracts
 
-- `dao-core` - the core module for DAOs.
+- `dao-dao` - the core module for DAOs.
```
Comment (NoahSaso) [inline contracts/README.md:3]:
```suggestion
- `dao-dao-core` - the core module for DAOs.
```

File: contracts/dao-dao/Cargo.toml (hunk 3 above)
Comment (NoahSaso) [reply contracts/dao-dao/Cargo.toml:2]: `dao-dao-core` 🙏

File: contracts/dao-dao/README.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# dao-core
+# dao-dao
```
Comment (NoahSaso) [inline contracts/dao-dao/README.md:1]:
```suggestion
# dao-dao-core
```

File: contracts/dao-dao/src/contract.rs [hunk 7]
```diff
@@ -5,25 +5,29 @@ use cosmwasm_std::{
...
 use crate::state::{
-    Config, ProposalModule, ProposalModuleStatus, ACTIVE_PROPOSAL_MODULE_COUNT, ADMIN, CONFIG,
-    CW20_LIST, CW721_LIST, ITEMS, NOMINATED_ADMIN, PAUSED, PROPOSAL_MODULES, SUBDAO_LIST,
-    TOTAL_PROPOSAL_MODULE_COUNT, VOTING_MODULE,
+    ACTIVE_PROPOSAL_MODULE_COUNT, ADMIN, CONFIG, CW20_LIST, CW721_LIST, ITEMS, NOMINATED_ADMIN,
+    PAUSED, PROPOSAL_MODULES, SUBDAO_LIST, TOTAL_PROPOSAL_MODULE_COUNT, VOTING_MODULE,
 };
 
-pub(crate) const CONTRACT_NAME: &str = "crates.io:dao-core";
+pub(crate) const CONTRACT_NAME: &str = "crates.io:dao-dao";
```
Comment (NoahSaso) [inline contracts/dao-dao/src/contract.rs:30]:
```suggestion
pub(crate) const CONTRACT_NAME: &str = "crates.io:dao-dao-core";
```

File: packages/cw-hooks/src/lib.rs [hunk 8]
```diff
@@ -43,7 +43,7 @@ impl<'a> Hooks<'a> {
 
     pub fn remove_hook(&self, storage: &mut dyn Storage, addr: Addr) -> Result<(), HookError> {
         let mut hooks = self.0.load(storage)?;
-        if let Some(p) = hooks.iter().position(|x| x == &addr) {
+        if let Some(p) = hooks.iter().position(|h| h == addr) {
```
Comment (NoahSaso) [inline packages/cw-hooks/src/lib.rs:46]: why is the `&` no longer needed?

File: packages/cw-hooks/src/lib.rs [hunk 9]
```diff
@@ -33,7 +33,7 @@ impl<'a> Hooks<'a> {
 
     pub fn add_hook(&self, storage: &mut dyn Storage, addr: Addr) -> Result<(), HookError> {
         let mut hooks = self.0.may_load(storage)?.unwrap_or_default();
-        if !hooks.iter().any(|h| h == &addr) {
+        if !hooks.iter().any(|h| h == addr) {
```
Comment (NoahSaso) [inline packages/cw-hooks/src/lib.rs:36]: why is the `&` no longer needed?

File: packages/dao-pre-propose-base/src/msg.rs [hunk 10]
```diff
@@ -98,7 +98,7 @@ where
     /// associated with. Returns `Addr`.
     #[returns(cosmwasm_std::Addr)]
     ProposalModule {},
-    /// Gets the DAO (cw-dao-core) module this contract is associated
+    /// Gets the DAO (dao-dao) module this contract is associated
```
Comment (NoahSaso) [inline packages/dao-pre-propose-base/src/msg.rs:101]:
```suggestion
    /// Gets the DAO (dao-dao-core) module this contract is associated
```

File: packages/dao-pre-propose-base/src/state.rs [hunk 11]
```diff
@@ -21,7 +21,7 @@ pub struct Config {
 pub struct PreProposeContract<InstantiateExt, ExecuteExt, QueryExt, ProposalMessage> {
     /// The proposal module that this module is associated with.
     pub proposal_module: Item<'static, Addr>,
-    /// The DAO (cw-dao-core module) that this module is associated
+    /// The DAO (dao-dao module) that this module is associated
```
Comment (NoahSaso) [inline packages/dao-pre-propose-base/src/state.rs:24]:
```suggestion
    /// The DAO (dao-dao-core module) that this module is associated
```

**Review by NoahSaso — CHANGES_REQUESTED** [review]
sick!! just a few notes

File: contracts/README.md (hunk 5 above)
Comment (JakeHartnell) [reply contracts/README.md]: Done.

File: contracts/dao-dao/README.md (hunk 6 above)
Comment (JakeHartnell) [reply contracts/dao-dao/README.md]: done

File: packages/cw-hooks/src/lib.rs (hunk 9 above)
Comment (JakeHartnell) [reply packages/cw-hooks/src/lib.rs:36]: New rust version.

File: packages/cw-hooks/src/lib.rs (hunk 8 above)
Comment (JakeHartnell) [reply packages/cw-hooks/src/lib.rs:46]: New rust stable version.

File: Cargo.toml (hunk 4 above)
Comment (JakeHartnell) [reply Cargo.toml]: Done!

**Comment by JakeHartnell** [comment]
@NoahSaso, have another look, I think we are good.

**Review by bekauz — APPROVED** [review]
lgtm!

**Review by NoahSaso — APPROVED** [review]

**Pull request merged**